syntax = "proto3";

import "brewblox.proto";
import "nanopb.proto";

package blox;

message SetpointSensorPair {
    option (nanopb_msgopt).msgid = 303;

    uint32 setpointId = 1 [(brewblox).link = Setpoint, (nanopb).int_size = IS_16];
    uint32 sensorId = 2 [(brewblox).link = TempSensor, (nanopb).int_size = IS_16];

    bool setpointValid = 3 [(brewblox).readonly = true];
    bool sensorValid = 4 [(brewblox).readonly = true];

    sint32 setpointValue = 5 [(brewblox).unit = Temp, (brewblox).scale = 4096, (nanopb).int_size = IS_32, (brewblox).readonly = true];
    sint32 sensorValue = 6 [(brewblox).unit = Temp, (brewblox).scale = 4096, (nanopb).int_size = IS_32, (brewblox).readonly = true];
}
